### PR TITLE
NodeLabelPriority as Score Plugin

### LINKED ISSUE #84456: NodeLabelPriority as Score Plugin
**What would you like to be added**:
convert NodeLabelPriority to a score plugin
https://github.com/kubernetes/kubernetes/blob/571cea0519eef38d77524f4074b3bd31bb7f5091/pkg/scheduler/algorithm_factory.go#L393-L402

**Why is this needed**:
part of #83554

**Additional Notes**
#84297 as an example for converting a custom predicate to a filter plugin. The work required here should be very similar to it.

/sig scheduling
/cc @ahg-g
/priority important-soon
/help

### PR DESCRIPTION
**Pull request by notpad**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #84456

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/algorithm_factory.go [hunk 1]
```diff
@@ -402,6 +406,8 @@ func RegisterCustomPriorityFunction(policy schedulerapi.PriorityPolicy, args *pl
 				},
 				Weight: policy.Weight,
 			}
+			// We do not allow specifying the name for custom plugins, see #83472
```
Comment (draveness) [inline pkg/scheduler/algorithm_factory.go]: nit: use http://issues.k8s.io/83472 instead

**Review by draveness — COMMENTED** [review]
/ok-to-test

Generally looks good

File: pkg/scheduler/framework/plugins/nodelabel/node_label.go [hunk 2]
```diff
@@ -37,25 +39,37 @@ type Args struct {
 	Labels []string `json:"labels,omitempty"`
 	// The boolean flag that indicates whether the labels should be present or absent from the node
 	Presence bool `json:"presence,omitempty"`
+
+	// The parameters that are used to configure the priority function
+	// The label that identify node "groups"
+	PreferenceLabel string `json:"preferenceLabel,omitempty"`
```
Comment (liu-cong) [inline pkg/scheduler/framework/plugins/nodelabel/node_label.go]: Do we know if the label/presence for the priority has anything to do with the labels/presence for the predicate? Or are they completely independent?

cc @ahg-g

Comment (ahg-g) [reply pkg/scheduler/framework/plugins/nodelabel/node_label.go]: we are going to change our approach with regards to custom predicates/priorities, @liu-cong will have an update soon.

File: pkg/scheduler/framework/plugins/nodelabel/node_label.go [hunk 3]
```diff
@@ -37,25 +39,37 @@ type Args struct {
...
 }
 
 // New initializes a new plugin and returns it.
-func New(plArgs *runtime.Unknown, _ framework.FrameworkHandle) (framework.Plugin, error) {
+func New(plArgs *runtime.Unknown, handle framework.FrameworkHandle) (framework.Plugin, error) {
 	args := &Args{}
 	if err := framework.DecodeInto(plArgs, args); err != nil {
 		return nil, err
 	}
+	prioritize, _ := priorities.NewNodeLabelPriority(args.PreferenceLabel, args.PreferenceLabelPresence)
```
[the remark this answers is not in the document]
Comment (notpad) [reply pkg/scheduler/framework/plugins/nodelabel/node_label.go]: done

File: pkg/scheduler/algorithm_factory.go (hunk 1 above)
Comment (notpad) [reply pkg/scheduler/algorithm_factory.go]: done

**Review by draveness — APPROVED** [review]
/lgtm

**Comment by ahg-g** [comment]
/hold

@liu-cong will have an update on how we should be dealing with custom predicates/priorities. Basically we need to allow more than one to be defined, but merge the labels and register a single plugin.

**Pull request closed without merging**